### PR TITLE
Update debpkg container image to ubuntu 18

### PR DESCRIPTION
**Pull request by am11**
Ubuntu 14 is out of support. Update to latest available debpkg container.

### CODE REVIEW

**Comment by teo-tsirpanis** [comment]
Why not even further like 20 or even 22?

**Comment by am11** [comment]
We currently do not have images higher than Ubuntu 18, with `debpkg` prereqs installed. I have just updated to the [latest available](https://raw.githubusercontent.com/dotnet/versions/main/build-info/docker/image-info.dotnet-dotnet-buildtools-prereqs-docker-main.json) image in order to abandon Ubuntu 14.

Good point tough. We can send a PR to build one with 22.04 (similar to https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/66216d372d2f80144b6a53535a46f2c7ced0d17e/src/ubuntu/18.04/debpkg/Dockerfile), that will buy us couple of more years.

**Comment by am11** [comment]
Main with Ubuntu 14.04: https://dev.azure.com/dnceng/public/_build/results?buildId=1730882&view=logs&j=7f5ce049-9fcc-50f8-e470-d8e52687d56a&t=d71fcd51-a472-5d5e-3f5a-e482fbd6ea4b 

PR with Ubuntu 18.04: https://dev.azure.com/dnceng/public/_build/results?buildId=1731561&view=logs&j=7f5ce049-9fcc-50f8-e470-d8e52687d56a&t=d71fcd51-a472-5d5e-3f5a-e482fbd6ea4b

**Comment by jkoritzinsky** [comment]
It looks good to me, but I don't have time to pull down packages to verify. @NikolaMilosavljevic can you take a closer look?

**Comment by NikolaMilosavljevic** [comment]
@am11 have you verified that packages are being produced? When I attempted similar change last year, installer build succeeded but did not produce any DEB packages. Perhaps this is a newer docker image and works well.

**Comment by am11** [comment]
@NikolaMilosavljevic, in the CI logs:

```
  dotnet-runtime-deps-debian -> /root/runtime/artifacts/packages/Release/Shipping/dotnet-runtime-deps-7.0.0-ci-x64.deb

Build succeeded.
    0 Warning(s)
    0 Error(s)
```

in both main and PR. Do you have any specific validation method in mind? I can build them locally and compare file sizes in the tree, for example.

**Comment by NikolaMilosavljevic** [comment]
> @NikolaMilosavljevic, from the CI logs, we see:
> 
> ```
>   dotnet-runtime-deps-debian -> /root/runtime/artifacts/packages/Release/Shipping/dotnet-runtime-deps-7.0.0-ci-x64.deb
> 
> Build succeeded.
>     0 Warning(s)
>     0 Error(s)
> ```
> 
> in both main and PR legs. Do you have any specific validation method in mind? I can build them locally and compare file sizes in the tree, for example.

That's great. The only validation I would suggest, would be to download produced DEBs (or build them locally) and try to install on a repro machine/container.

**Comment by am11** [comment]
Packages (.deb) are produced but they are not uploaded to build artifacts (only log files are uploaded). This is the case with both:

* `main` branch with Ubuntu 14.04: https://dev.azure.com/dnceng/public/_build/results?buildId=1730882&view=logs&j=7f5ce049-9fcc-50f8-e470-d8e52687d56a&t=d71fcd51-a472-5d5e-3f5a-e482fbd6ea4b 

* and this PR with Ubuntu 18.04: https://dev.azure.com/dnceng/public/_build/results?buildId=1731561&view=logs&j=7f5ce049-9fcc-50f8-e470-d8e52687d56a&t=d71fcd51-a472-5d5e-3f5a-e482fbd6ea4b

**Comment by am11** [comment]
@ViktorHofer, @hoyosjs, do you know if we can get pipeline to upload artifacts (one off), so we can verify the produced package `dotnet-runtime-deps-7.0.0-ci-x64.deb` from `Package Runtime Deps, Runtime, Framework Packs installers - deb` [step](https://dev.azure.com/dnceng/public/_build/results?buildId=1731561&view=logs&j=7f5ce049-9fcc-50f8-e470-d8e52687d56a&t=d71fcd51-a472-5d5e-3f5a-e482fbd6ea4b) is valid?

(ideally we should have a build step to install the produced deb and rpm in clean containers to automate this verification process..)

**Comment by ViktorHofer** [comment]
Probably by removing this if: https://github.com/dotnet/runtime/blob/e3860440c151897968eca2227ae98c1912580c44/eng/pipelines/installer/jobs/steps/upload-job-artifacts.yml#L10.

**Comment by am11** [comment]
Thank you. I have inverted the condition temporarily so we can test the *.deb artifacts.

**Comment by am11** [comment]
@NikolaMilosavljevic, the intermediate artifacts archive that has deb file was uploaded by the last run here: https://dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_apis/build/builds/1756291/artifacts?artifactName=IntermediateArtifacts&api-version=7.0&%24format=zip (size: 7.2 GB)

image


I have successfully installed the .deb file on Ubuntu 22.04, 20.04 and 18.04 and executed a few commands, it works.
e.g. on 20.04:
```sh
$ cd extract_dir_of_selected_files_from_screenshot
$ docker run -v$(pwd):/assets1 -it ubuntu:20.04

# inside the container
$ apt update && apt install -y libicu-dev libkrb5-dev
$ cd /assets1
$ dpkg -i *.deb

Selecting previously unselected package dotnet-apphost-pack-7.0.
(Reading database ... 10510 files and directories currently installed.)
Preparing to unpack dotnet-apphost-pack-7.0.0-ci-x64.deb ...
Unpacking dotnet-apphost-pack-7.0 (7.0.0~ci-1) ...
Selecting previously unselected package dotnet-host.
Preparing to unpack dotnet-host-7.0.0-ci-x64.deb ...
Unpacking dotnet-host (7.0.0~ci-1) ...
Selecting previously unselected package dotnet-hostfxr-7.0.
Preparing to unpack dotnet-hostfxr-7.0.0-ci-x64.deb ...
Unpacking dotnet-hostfxr-7.0 (7.0.0~ci-1) ...
Selecting previously unselected package dotnet-runtime-7.0.
Preparing to unpack dotnet-runtime-7.0.0-ci-x64.deb ...
Unpacking dotnet-runtime-7.0 (7.0.0~ci-1) ...
Selecting previously unselected package dotnet-runtime-deps-7.0.
Preparing to unpack dotnet-runtime-deps-7.0.0-ci-x64.deb ...
Unpacking dotnet-runtime-deps-7.0 (7.0.0~ci-1) ...
Selecting previously unselected package dotnet-targeting-pack-7.0.
Preparing to unpack dotnet-targeting-pack-7.0.0-ci-x64.deb ...
Unpacking dotnet-targeting-pack-7.0 (7.0.0~ci-1) ...
Setting up dotnet-apphost-pack-7.0 (7.0.0~ci-1) ...
Setting up dotnet-host (7.0.0~ci-1) ...
Setting up dotnet-hostfxr-7.0 (7.0.0~ci-1) ...
Setting up dotnet-runtime-deps-7.0 (7.0.0~ci-1) ...
Setting up dotnet-targeting-pack-7.0 (7.0.0~ci-1) ...
Setting up dotnet-runtime-7.0 (7.0.0~ci-1) ...

$ dotnet --info

global.json file:
  Not found

Host:
  Version:      7.0.0-ci
  Architecture: x64
  Commit:       7d2ad9838e

.NET SDKs installed:
  No SDKs were found.

.NET runtimes installed:
  Microsoft.NETCore.App 7.0.0-ci [/usr/share/dotnet/shared/Microsoft.NETCore.App]

Download .NET:
  https://aka.ms/dotnet-download

Learn about .NET Runtimes and SDKs:
  https://aka.ms/dotnet/runtimes-sdk-info
```

**Review by ViktorHofer — APPROVED** [review]

File: eng/pipelines/installer/jobs/base-job.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ parameters:
   pgoType: ''
 
   packageDistroList:
-  - image: mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-debpkg-20220408155542-cfdd435
+  - image: mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-debpkg-20220421191645-cfdd435
```
Comment (am11) [inline eng/pipelines/installer/jobs/base-job.yml:26]:
```suggestion
  - image: mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-debpkg-20220504035737-cfdd435
```

**Comment by am11** [comment]
Resolved merge conflicts..
@hoyosjs I see that you have already updated it to Ubuntu 18 in https://github.com/dotnet/runtime/commit/6bc1c947068089879d448a67aeedd59caaa9d601. I have bumped to the latest prerequisite ref.

**Review by hoyosjs — APPROVED** [review]

**Review by ViktorHofer — APPROVED** [review]

**Pull request merged**